components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/utils/XMLUtils.java

```diff
@@ -32,15 +38,20 @@
     /**
      * Build a Document from a file
      *
-     * @param xmlFile file
+     * @param xmlFilePath file
      * @return a Document object
      * @throws Exception if an error occurs
      */
-    public static Document buildDocumentFromFile(File xmlFile) throws Exception {
+    //public static Document buildDocumentFromFile(File xmlFile)
+    public static Document buildDocumentFromFile(Path xmlFilePath)
+            throws ParserConfigurationException, IOException, SAXException {
+        InputStream inputStream =Files.newInputStream(xmlFilePath);
+
         DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
         DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
-        Document doc = dBuilder.parse(xmlFile);
+        Document doc = dBuilder.parse(inputStream);
         doc.getDocumentElement().normalize();
+
```

Comment:
don't we need to close the input stream here ? or is it automatically getting closed at the document builder ?